tinygrad/tensor.py

```diff
@@ -553,13 +553,16 @@ def _broadcasted(self, fxn:Type[Function], other:Union[Tensor, float], reverse:b
   def add(self, x:Union[Tensor, float], reverse=False) -> Tensor: return self._broadcasted(mlops.Add, x, reverse) if x.__class__ is Tensor or x else self
   def sub(self, x:Union[Tensor, float], reverse=False) -> Tensor: return self._broadcasted(mlops.Sub, x, reverse) if x.__class__ is Tensor or x or reverse else self
   def mul(self, x:Union[Tensor, float], reverse=False) -> Tensor: return self._broadcasted(mlops.Mul, x, reverse) if x.__class__ is Tensor or x != 1.0 else self
+  def div(self, x:Union[Tensor, float], reverse=False) -> Tensor: return self._broadcasted(mlops.Div, x, reverse) if x.__class__ is Tensor or reverse or not x else self.mul(1/x)
   def pow(self, x:Union[Tensor, float], reverse=False) -> Tensor:
+    a = self
     if x.__class__ is not Tensor and not reverse:
       # simple pow identities
-      if x == 2.0: return self*self
-      if x == -1.0: return 1/self
-    return self._broadcasted(mlops.Pow, x, reverse) if x.__class__ is Tensor or x != 1.0 or reverse else self
-  def div(self, x:Union[Tensor, float], reverse=False) -> Tensor: return self._broadcasted(mlops.Div, x, reverse) if x.__class__ is Tensor or reverse or not x else self.mul(1/x)
+      if x < 0: return (1.0/self).pow(-x)
+      if x == 2.0: return a*a
+      if x == 1.0: return a
+      if x == 0.5: return a.sqrt()
+    return self.log().mul(x).exp() if not reverse or isinstance(x, Tensor) else self.mul(log(x)).exp()
```

Comment:
Is this handling reverse correctly?

Reply:
Yes. It passes test_pow and test_pow_const. It would fail if you tried to run Tensor.pow(Tensor, True), though. That is extremely abnormal usage of pow but do you want it to work?

Reply:
err, is there no way rpow can ever be called like that? it used to work, didn't it?

Reply:
rpow only gets called when you do `float**Tensor` so it will never get called for `Tensor**Tensor`
it did used to work without having to add anything extra since it could just pass the `reverse` argument to the `self._broadcasted` but it would take about two more lines to make it work here